astrolabe/commands.py

```diff
@@ -0,0 +1,138 @@
+from time import time, sleep
+
+from astrolabe.utils import assert_subset
+from atlasclient import AtlasApiError
+
+
+def get_one_organization_by_name(*, client, organization_name):
+    all_orgs = client.orgs.get().data
+    for org in all_orgs.results:
+        if org.name == organization_name:
+            return org
+    raise AtlasApiError('Resource not found.')
+
+
+def ensure_project(*, client, group_name, organization_id):
+    try:
+        return client.groups.post(
+            name=group_name, orgId=organization_id).data
+    except AtlasApiError as exc:
+        if exc.error_code == 'GROUP_ALREADY_EXISTS':
+            return client.groups.byName[group_name].get().data
+        else:
+            raise
+
+
+def ensure_admin_user(*, client, group_id, username, password):
+    user_details = {
+        "groupId": group_id,
+        "databaseName": "admin",
+        "roles": [{
+            "databaseName": "admin",
+            "roleName": "atlasAdmin"}],
+        "username": username,
+        "password": password}
+
+    try:
+        return client.groups[group_id].databaseUsers.post(**user_details).data
+    except AtlasApiError as exc:
+        if exc.error_code == "USER_ALREADY_EXISTS":
+            username = user_details.pop("username")
+            return client.groups[group_id].databaseUsers.admin[username].patch(
+                **user_details).data
+        else:
+            raise
+
+
+def ensure_connect_from_anywhere(*, client, group_id,):
+    ip_details_list = [{"cidrBlock": "0.0.0.0/0"}]
+    client.groups[group_id].whitelist.post(json=ip_details_list)
+
+
+def get_cluster_state(client, group_name, cluster_name):
```

Comment:
Why do some functions have `*` and some don't?

Reply:
This was old code and is now removed. For the most part, I have tried to have functions be kwarg only, unless doing so would make things exceptionally verbose. I find that doing this improves the readability of code and reduces chances of errors.